src/opendr/perception/pose_estimation/hr_pose_estimation/high_resolution_learner.py

```diff
@@ -0,0 +1,610 @@
+        pool_img = pool_img.unsqueeze(0)
+        pool_img = torch.nn.functional.avg_pool2d(pool_img, kernel)
+        pool_img = pool_img.squeeze(0).permute(1, 2, 0).cpu().float().numpy()
+        return pool_img
+
+    def fit(self, dataset, val_dataset=None, logging_path='', silent=True, verbose=True):
+        """This method is not used in this implementation."""
+
+        raise NotImplementedError
+
+    def optimize(self, target_device):
+        """This method is not used in this implementation."""
+
+        raise NotImplementedError
+
+    def reset(self):
+        """This method is not used in this implementation."""
+        return NotImplementedError
+
+    def save(self, path):
+        """This method is not used in this implementation."""
+        return NotImplementedError
+
+    def eval(self, dataset,  silent=False, verbose=True, use_subset=True, subset_size=250, upsample_ratio=4,
+             images_folder_name="val2017", annotations_filename="person_keypoints_val2017.json"):
+        """
+                This method is used to evaluate a trained model on an evaluation dataset.
+
+                :param dataset: object that holds the evaluation dataset.
+                :type dataset: ExternalDataset class object or DatasetIterator class object
+                :param silent: if set to True, disables all printing of evalutaion progress reports and other
+                    information to STDOUT, defaults to 'False'
+                :type silent: bool, optional
+                :param verbose: if set to True, enables the maximum verbosity, defaults to 'True'
+                :type verbose: bool, optional
+                :param use_subset: If set to True, a subset of the validation dataset is created and used in
+                    evaluation, defaults to 'True'
+                :type use_subset: bool, optional
+                :param subset_size: Controls the size of the validation subset, defaults to '250'
+                :type subset_size: int, optional
+                param upsample_ratio: Defines the amount of upsampling to be performed on the heatmaps and PAFs
+                    when resizing,defaults to 4
+                :type upsample_ratio: int, optional
+                :param images_folder_name: Folder name that contains the dataset images. This folder should be contained
+                in the dataset path provided. Note that this is a folder name, not a path, defaults to 'val2017'
+                :type images_folder_name: str, optional
+                :param annotations_filename: Filename of the annotations json file. This file should be contained in the
+                    dataset path provided, defaults to 'pesron_keypoints_val2017.json'
+                :type annotations_filename: str, optional
+
+                :returns: returns stats regarding evaluation
+                :rtype: dict
+                """
+
+        data = super(HighResolutionPoseEstimationLearner,
```

Comment:
Local IDE shows quite big yellow lines for `Unresolved attribute reference '_LightweightOpenPoseLearner__prepare_val_dataset' for class 'LightweightOpenPoseLearner'`. `NOQA` disables it as it looks bad.
```suggestion
        data = super(HighResolutionPoseEstimationLearner,  # NOQA
```